model/src/main/kotlin/io/spine/validation/required/CombinationParser.kt

```diff
@@ -0,0 +1,159 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation.required
+
+import io.spine.option.RequireOption
+import io.spine.protodata.Compilation
+import io.spine.protodata.ast.Field
+import io.spine.protodata.ast.File
+import io.spine.protodata.ast.MessageType
+import io.spine.protodata.ast.name
+import io.spine.protodata.ast.qualifiedName
+import io.spine.protodata.check
+import io.spine.validation.FieldCombination
+import io.spine.validation.REQUIRE
+import io.spine.validation.fieldCombination
+import io.spine.validation.required.RequiredFieldSupport.isSupported
+
+/**
+ * Parses and validates field combinations specified in the given [RequireOption].
+ *
+ * The class ensures the following conditions:
+ *
+ * 1. Each specified field is declared in the [message] type.
+ * 2. Each field has a [compatible][RequiredFieldSupport] field type.
+ * 3. Each combination has a unique set of fields.
+ * 4. Each combination is unique.
+ *
+ * A compilation error is reported in case of violation of any condition.
+ */
+internal class CombinationParser(
+    option: RequireOption,
+    private val message: MessageType,
+    private val file: File
+) {
+    private val specifiedFields = option.fields
+    private val messageFields = message.fieldList
+    private val messageFieldNames = messageFields
+        .map { it.name.value }
+        .toSet()
+
+    /**
+     * Returns a list of parsed [FieldCombination].
+     */
+    fun combinations(): List<FieldCombination> {
+        val combinations = specifiedFields.split(FIELDS_DELIMITER)
+            .map(::toCombination)
+            .run { checkCombinationsUnique(this) }
+        return combinations.map {
+            val astFields = it.fields
+                .map(::toAstField)
+                .onEach(::checkFieldType)
+            fieldCombination {
+                field.addAll(astFields)
+            }
+        }
+    }
+
+    private fun toCombination(definition: String): Combination {
```

Comment:
The `Combination` reminds me on the name of a girly pop group, which never sounded like a serious (in the sense of quality) music. 

Can we be more specific in this context, please?